### PR TITLE
fix bugs!

### LINKED ISSUE #481: No Age Group Section: I see all children without age groups in a section
Currently:
Children with no age group are not displayed in the roster.

Ideal:
Children without agegroup are displayed in "no age group" section
Screen Shot 2020-10-06 at 2 40 31 PM

### LINKED ISSUE #492: After editing a child's profile and saving, the system generates 2 alerts and 1 is an error messaging saying the information just edited is still missing
Screen Shot 2020-10-02 at 5 14 13 PM
– After adding in needed info the error message persists. After hitting save again, the error message goes away.

### LINKED ISSUE #508: Old missing info alert on the roster needs to clear when a new file is uploaded.
Screen Shot 2020-10-06 at 2 19 04 PM
When I see a missing info alert on the roster, I go to batch upload another file. When I upload that file and return to the roster, the old missing info alert is still there. 

See also #451 for the confirmation alert that should show after batch uploading a file.

### LINKED ISSUE #545: Roster empty state alert shows up when refreshing a not empty roster.

### LINKED ISSUE #556: cant add agegroup to enrollment

### PR DESCRIPTION
**Pull request by jlhogan**
closes #545 
closes #556
closes #508 
closes #481 
closes #492 
closes new bug i introduced by merging a diff think that broke file upload :)

### CODE REVIEW

File: client/src/containers/Roster/Roster.tsx [hunk 1]
```diff
@@ -56,12 +58,19 @@ const Roster: React.FC = () => {
...
+    }
+
     if (numberOfChildrenWithErrors) {
       setAlerts([...alerts, childrenWithErrorsAlert]);
     }
-  }, [numberOfChildrenWithErrors]);
+  }, [children.loading, numberOfChildrenWithErrors]);
 
-  const childrenByAgeGroup: { [key in AgeGroup]?: Child[] } = {};
+  const NoAgeGroup = 'No age group';
```
Comment (kmagida110) [inline client/src/containers/Roster/Roster.tsx:71]: Should this be in the model or will this not actually be pushed to the DB?

**Review by kmagida110 — APPROVED** [review]
LGTM, one question for my own sake

File: client/src/containers/EditRecord/EditRecord.tsx [hunk 2]
```diff
@@ -48,41 +49,47 @@ const EditRecord: React.FC = () => {
   useEffect(() => {
     apiGet(`children/${childId}`, {
       accessToken,
-    }).then((_rowData) => setRowData(_rowData));
+    }).then((updatedChild) => {
+      children.addOrUpdateRecord(updatedChild);
+
+      // On initial fetch, refetch = 0 AND we do not want to create alerts
+      if (refetch) {
```
Comment (JordanGuinn) [inline client/src/containers/EditRecord/EditRecord.tsx:56]: nitty nit: thoughts on renaming this to `fetchCount` or something?

**Review by JordanGuinn — APPROVED** [review]
LGTM 👌🏽👌🏽

**Review by mmazanec22 — APPROVED** [review]

File: .circleci/config.yml [hunk 3]
```diff
@@ -48,6 +48,12 @@ jobs:
     steps:
       - attach_workspace:
           at: .
+      - run:
+          name: Run npm audit for back-end dependencies
+          command: npm i --package-lock-only && npm audit
+      - run:
+          name: Run npm audit for front-end dependencies
+          command: npm i --package-lock-only && npm audit
```
[the remark this answers is not in the document]
Comment (JordanGuinn) [reply .circleci/config.yml:56]: Ooooooo, I didn't realize that was a yarn specific issue 😬 In that case, yeah if npm works then screw it

**Pull request merged**